clients/src/main/java/org/apache/kafka/clients/admin/internals/CoordinatorKey.java

```diff
@@ -24,7 +24,7 @@
     public final String idValue;
```

Comment:
Since we are now going to be returning a `CoordinatorKey` type in some of the results we return (eg: `DescribeConsumerGroupResult`, `DeleteConsumerGroupResult`), perhaps we should move this out of the the `internals` package and instead put it in the `clients/src/main/java/org/apache/kafka/clients/admin` directory instead.

Reply:
This package is part of the public API. I'm not sure we want to expose `CoordinatorKey` which is currently only used internally.
That said, [KIP-692](https://cwiki.apache.org/confluence/display/KAFKA/KIP-692%3A+Make+AdminClient+value+object+constructors+public) proposes exposing the constructors of all the `*Result` classes. Maybe that discussion should happen there?

Reply:
Yeah I can definitely propose that in the discussion thread there. As I was working on my implementation on KIP-709 for batching the `listConsumerGroupsOffsets` API, I saw the need to make this class public since it will be needed to handle the `ListConsumerGroupsOffsetsResult` class properly. But I'll take this discussion to the email thread/the PR I'll open shortly.

Reply:
@skaundinya15 I don't understand why we would want to return `CoordinatorKey` in some results. Do we really have to? It seems that we could unwrap it depending on the context and only expose what the user need (e.g. group id).

Personally, I believe that we should strive for keeping all the constructors of `*Result` classes package private for the time being until KIP-692 is accepted, if it will.

Reply:
@dajac Ah I see the issue now, you're right - we don't need to return `CoordinatorKey` as part of the result. I was mistaken here - we can just keep the original method signature and convert the internal `CoordinatorKey` mapping to the `String` group id. Apologies for the confusion, the way that @mimaison has it here works just fine, I'll use a similar approach when implementing KIP-709.